sjasm/io_trd.cpp

```diff
@@ -195,8 +196,32 @@ int TRD_AddFile(char* fname, char* fhobname, int start, int length, int autostar
 			hdr[0x09] = (unsigned char)(start & 0xff);
 			hdr[0x0a] = (unsigned char)(start >> 8);
 		}
+}
+*/
+
+	memset(hdr,' ',9);
+	i = strlen(fhobname);
+	if (i > 1)
+	{
+		char *ext = strrchr(fhobname, '.');
+		if (ext && ext[1])
+		{
+			hdr[8] = ext[1];
+			*ext = '\0';
+			i = strlen(fhobname);
+		}
+	}
```

Comment:
Maybe warning about truncating extension and/or name can be issued here? If that makes sense... like
`if (8 < i || (ext && ext[1] && ext[2])) { Warning("Invalid filename in trdos, will get truncated", fhobname); }` (and needs no change to fhobname, will add comment above how to achieve that)

Reply:
Strictly speaking, you are probably right in that there should be at least some indication about it, and this was also my initial idea. However, I think this is ideally decided/changed together with other similar save functions like SAVETAP (it doesn't make sense if one function produces warnings and the other does not), which is outside of the scope of this fix.